toptal/picasso
# fix: fix unit test errors/warnings

[FX-2598]

### Description

Fixed all unit test console errors caused by unhandled async work or state changes
Provided some `waitFor` and `act` to be able to fix.
I left only two warnings which needs more research or refactoring to solve

### How to test

- when you run `yarn test:unit`, you will only see two warnings(see screenshots)

### Screenshots
<img width="584" alt="Screen Shot 2022-04-13 at 11 49 51" src="https://user-images.githubusercontent.com/7733167/163168695-72487f35-6b44-40f2-976c-aedc2f436bdc.png">
<img width="1168" alt="Screen Shot 2022-04-13 at 11 50 02" src="https://user-images.githubusercontent.com/7733167/163168699-9ede36dd-9bde-4b8d-89da-3db7d74d9023.png">

### Review


<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>
